src/snowflake/snowpark/_internal/utils.py

```diff
@@ -846,8 +846,9 @@ def infer_ast_enabled_from_global_sessions(func: Callable) -> bool:  # pragma: n
                 f"Could not retrieve default session "
                 f"for function {func.__qualname__}, capturing AST by default."
             )
-            # session has not been created yet. To not lose information, always encode AST.
-            return True  # noqa: B012
+            # session has not been created yet, do not encode AST be default.
+            # TODO: flip this to True when we go GA.
+            return False  # noqa: B012
```

Comment:
If we're moving away from hard-coded `True`, the value should come from an environmental variable, or some other suitable global configuration source. We don't want to be in a place where the feature is untestable while under development/opt-in mode.

Reply:
Why would it not be testable if user has a session and they have oped-in such that `session.ast_enabled` returns true.

This is the case when a valid session is not found, and we default to assuming that ast is enabled. I say, we default to disabled.